graphql_compiler/tests/test_helpers.py

```diff
@@ -407,3 +408,8 @@ def get_test_macro_registry():
     for graphql, args in valid_macros:
         register_macro_edge(macro_registry, graphql, args)
     return macro_registry
+
+
+def get_required_macro_edge_directives():
+    """Return a tuple containing the directives necessary for defining a macro edge"""
+    return (MacroEdgeDefinitionDirective, MacroEdgeTargetDirective)
```

Comment:
Great idea materializing this tuple. But it might be better to move it `macros/directives.py`, as a constant. This is mostly for consistency with the codebase (that's where people would expect to find it), but also because this information is used outside of tests too.

Reply:
Fixed in commit "Address comments". I added sets in macro_edges/directives.py for allowed & disallowed directives in a macro, as well as directives necessary for defining a macro.